org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/references/utils/XMLReferencesSearcher.java

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+* Copyright (c) 2023 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.extensions.references.utils;
+
+import java.util.List;
+
+import org.eclipse.lemminx.extensions.references.settings.XMLReferenceExpression;
+
+/**
+ * The XML reference searcher.
+ * 
+ * @author Angelo ZERR
+ *
+ */
+public class XMLReferencesSearcher {

```

Comment:
This class doesn't actually perform the search, so maybe it's better to call it something like `XMLReferencesSearchParams`.

Reply:
I rename with `XMLReferencesSearchContext`